zerver/views/upload.py

```diff
@@ -283,3 +284,13 @@ def upload_file_backend(request: HttpRequest, user_profile: UserProfile) -> Http
 
     uri = upload_message_attachment_from_request(user_file, user_profile, file_size)
     return json_success(request, data={"uri": uri})
+
+
+def get_tusd_file_data(
+    request: HttpRequest, user_profile: UserProfile, file_id_str: str
+) -> HttpResponse:
+    try:
+        file = Attachment.objects.get(tus_file_id=file_id_str)
+        return json_success(request, {"url": "/user_uploads/" + file.path_id})
+    except (Attachment.DoesNotExist, ValidationError):
+        raise JsonableError(_("File not found."))
```

Comment:
Can you explain why this API endpoint is required? I had imagined that clients wouldn't need to know about TUSD in order to download files, only to upload them in chunks, so it's not obvious to me what this is for.

Probably we should document the answer in a docstring for this function.